js/views/search/Search.js

```diff
@@ -152,16 +157,24 @@ export default class extends baseVw {
    * This will create a url with the term and other query parameters
    * @param {string} term
    */
-  processTerm(term) {
+  processTerm(term, reset) {
     this.term = term || '';
     // if term is false, search for *
     const query = `q=${encodeURIComponent(term || '*')}`;
     const page = `&p=${this.serverPage}&ps=${this.pageSize}`;
     const sortBy = this.sortBySelected ? `&sortBy=${encodeURIComponent(this.sortBySelected)}` : '';
     const network = `&network=${!!app.serverConfig.testnet ? 'testnet' : 'mainnet'}`;
-    let filters = $.param(this.filters);
-    filters = filters ? `&${filters}` : '';
-    const newURL = `${this.providerUrl}?${query}${network}${sortBy}${page}${filters}`;
+    // if the DOM is ready, use the form values. Otherwise use the values set in the constructor.
+    let filters = this.$filters ? this.$filters.serialize() : $.param(this.filterParams);
+    filters = !filters || reset ? '' : `&${filters}`;
+    const newURL = new URL(`${this.providerUrl}?${query}${network}${sortBy}${page}${filters}`);
+    const newParams = newURL.searchParams;
+    if (!newParams.has('nsfw')) {
+      newParams.append('nsfw', app.settings.get('showNsfw'));
+    }
+    if (!newParams.has('acceptedCurrencies')) {
+      newParams.append('acceptedCurrencies', app.serverConfig.cryptoCurrency);
```

Comment:
Didn't we fall on always sending the non-testnet value, in which case we should be using `getServerCurrency().code`?

Reply:
Good catch. I updated it in the initial filters and forgot to here.

Reply:
Actually, this should be removed, since we're using no value for acceptedCurrencies to mean search for all of them.